### ISSUE TITLE
Voice cloning - unexpected results

### ISSUE THREAD

**Issue by BrasD99** [serp-ai/bark-with-voice-clone#49]
Hi!

I'm trying to clone my voice from a wav file to get the pronunciation of the text in Russian according to https://github.com/serp-ai/bark-with-voice-clone/blob/main/clone_voice.ipynb. The main problem is that the resulting wav contains voice acting not in Russian. I am running the code on macbook m2 (CPU) ;( 

Am I missing something?

Here is my code:
import sys
sys.path.append('bark-voice-cloning-HuBERT-quantizer')

from encodec.utils import convert_audio
from hubert.hubert_manager import HuBERTManager
from hubert.pre_kmeans_hubert import CustomHubert
from hubert.customtokenizer import CustomTokenizer
from bark.api import generate_audio
from bark.generation import SAMPLE_RATE, preload_models, load_codec_model
import torchaudio
import torch
import os
import numpy as np
from scipy.io.wavfile import write as write_wav

device = torch.device('cpu')
model = load_codec_model(use_gpu=False)

hubert_manager = HuBERTManager()
hubert_manager.make_sure_hubert_installed()
hubert_manager.make_sure_tokenizer_installed()

# Load the HuBERT model
hubert_model = CustomHubert(checkpoint_path='data/models/hubert/hubert.pt').to(device)

# Load the CustomTokenizer model
tokenizer = CustomTokenizer.load_from_checkpoint('data/models/hubert/tokenizer.pth').to(device)  # Automatically uses the right layers

audio_filepath = '/Users/brasd99/Downloads/original.wav' # the audio you want to clone (under 13 seconds)
wav, sr = torchaudio.load(audio_filepath)
wav = convert_audio(wav, sr, model.sample_rate, model.channels)
wav = wav.to(device)

semantic_vectors = hubert_model.forward(wav, input_sample_hz=model.sample_rate)
semantic_tokens = tokenizer.get_token(semantic_vectors)

# Extract discrete codes from EnCodec
with torch.no_grad():
    encoded_frames = model.encode(wav.unsqueeze(0))
codes = torch.cat([encoded[0] for encoded in encoded_frames], dim=-1).squeeze()  # [n_q, T]

# move codes to cpu
codes = codes.cpu().numpy()
# move semantic tokens to cpu
semantic_tokens = semantic_tokens.cpu().numpy()

voice_filename = 'ru_speaker_1'

current_path = os.getcwd()
voice_name = os.path.join(current_path, f'{voice_filename}.npz')

np.savez(voice_name, fine_prompt=codes, coarse_prompt=codes[:2, :], semantic_prompt=semantic_tokens)

# Enter your prompt and speaker here
text_prompt = "Привет! Как твои дела, друг? [laughs]"

# download and load all models
preload_models(
    text_use_gpu=False,
    text_use_small=False,
    coarse_use_gpu=False,
    coarse_use_small=False,
    fine_use_gpu=False,
    fine_use_small=False,
    codec_use_gpu=False,
    force_reload=False
)

# simple generation
audio_array = generate_audio(text_prompt, history_prompt=voice_name, text_temp=0.7, waveform_temp=0.7)

# save audio
filepath = "out.wav" # change this to your desired output path
write_wav(filepath, SAMPLE_RATE, audio_array)

Here is resulting wav: https://drive.google.com/file/d/1rbYL4lxzRPIZD25bVS27e0EkFZpjsFsE/view?usp=sharing

**Comment by BrasD99** [comment]
And additionally: the path parameter in the preload_models method no longer exists, you need to update ipynb

**Comment by BrasD99** [comment]
Same result for a small wav file (less than 13 seconds)

**Comment by sergeykorablin** [comment]
Тебе удалось решить проблему? 
У меня тоже модель игнорирует русский текст и говорит ерунду на английском. Временами и английский тект игнорирует и выдаает что-то рандомное.

**Comment by BrasD99** [comment]
> Тебе удалось решить проблему? 
> 
> У меня тоже модель игнорирует русский текст и говорит ерунду на английском. Временами и английский тект игнорирует и выдаает что-то рандомное.

Нет, не получилось. Постоянно что-то говорит на английском, причем это никак не связано ни с заданным мной текстом, ни с голосом. И никто ничего не отвечает мне(

**Comment by BrasD99** [comment]
> Наверное надо в основном проекте тикет заводить https://github.com/suno-ai/bark , этот производный

Не думаю, что в основном надо, потому что не работает процесс клонирования голоса, который реализован здесь. Но попробовать, конечно, можно, раз уж тут тишина)

**Comment by sergeykorablin** [comment]
Почитал тикеты в основном проекте, там тоже люди жалуются на французский и другие языки и тоже тишина в ответ)
Похоже модель обучалась в основном на англ и на других языках недообучена, отсюда и результат такой.

**Comment by 16bc** [comment]
Bark тут ни при чём.
Проблема ещё до него, в неправильном получаемом .npz файле, а точнее в модели хуберта и токенизатора, которые языкозависимые. Suno-ai выложили npz, но нам неизвестно с помощью каких моделей их получили.
Здесь в hubert_manager фигурирует два пути:

```
https://dl.fbaipublicfiles.com/hubert/hubert_base_ls960.pt
GitMylo/bark-voice-cloning/quantifier_hubert_base_ls960_14.pth
```

Которые, судя по документации (https://github.com/gitmylo/bark-voice-cloning-HuBERT-quantizer), обучены на английском языке.
![изображение](https://github.com/serp-ai/bark-with-voice-clone/assets/60779263/4b34a907-e548-423b-bfeb-59e2dbae6c2b)

**Comment by devinschumacher** [comment]
@BrasD99 

We just launched our follow up to BARK -- an AI Voice Cloner Desktop App (runs locally, offline, super realistic voice cloning, data privacy, no coding needed)

👉 Check it out: https://github.com/serpapps/ai-voice-cloner
